src/llm/openai.rs

```diff
@@ -214,4 +303,14 @@ mod test {
         let response = client.generate(prompt).await.unwrap();
         assert!(response.to_string().to_lowercase().contains("berlin"));
     }
+
+    #[tokio::test]
+    async fn test_embeddings() {
+        println!("test_embeddings");
+        let client = OpenAI::new();
+        let content: Content = Content::String("This is a test".to_string());
+        let record = Record::new(content);
+        let res = client.generate_embedding(&record).await.unwrap();
+        assert!(res.data.embedding.len() > 0);
+    }
```

Comment:
The test should look something like this using the correct prompting functionality:
```rust
    #[tokio::test]
    async fn test_embeddings() {
        println!("test_embeddings");
        let client = OpenAI::new();
        let content = "This is a test".to_string();
        let res = client.generate_embedding(Box::new(content)).await.unwrap();
        assert!(res.data[0].embedding.len() > 0);
    }
```

Reply:
That looks so much better, Thanks!